packages/filesystem/src/browser/filesystem-frontend-contribution.ts

```diff
@@ -82,23 +86,40 @@ export class FileSystemFrontendContribution implements FrontendApplicationContri
     registerCommands(commands: CommandRegistry): void {
         commands.registerCommand(FileSystemCommands.UPLOAD, new FileSelection.CommandHandler(this.selectionService, {
             multi: false,
-            isEnabled: selection => this.canUpload(selection),
-            isVisible: selection => this.canUpload(selection),
+            isEnabled: selection => this.menuItemEnabled(selection),
+            isVisible: selection => this.menuItemVisible(selection),
             execute: selection => this.upload(selection)
         }));
     }
 
-    protected canUpload({ fileStat }: FileSelection): boolean {
-        return !environment.electron.is() && fileStat.isDirectory;
+    //the top-left corner menu should always be enabled
+    protected menuItemEnabled(selection: FileSelection | undefined): boolean {
```

Comment:
It'd be better if we were consistent with other `docstring` for methods.
The comment regarding that the menu item should always be enabled can be placed within the method.

Example:

https://github.com/theia-ide/theia/blob/f5161dbc4882b800e928a11cf8b0901d76e29eee/packages/cpp/src/browser/cpp-build-configurations.ts#L74-L79